### DISCUSSION TITLE
Default for a null-only value

### DISCUSSION THREAD

**Discussion by dlampa** [mikefarah/yq#909] [Q&A] [answered]
Hi,

I have a key with either `true` or `false` assigned as a value. What would be the best way to return a default value for the key only if the key is not specified (i.e. null)?

Thanks!

**Comment by mikefarah** [comment]
Only way I can think of (cause I haven't yet added if..then..else into yq) is something like this:

```
yq e '(select(has("blah")) // {"blah": "things"}) | .blah
```

**Comment by mikefarah** [comment, marked as answer]
```
yq e '(select(has("blah") and .blah != null) // {"blah": "things"}) | .blah' examples/data1.yaml
```

If you want to use a default value when .blah is null

**Reply by dlampa** [reply]
Thanks, this worked perfectly!

**Reply by merlinrabens** [reply]
Thanks, exactly what I was looking for! Thanks @mikefarah
